app.go

```diff
@@ -0,0 +1,77 @@
+package supervisor
+
+import (
+	"github.com/caddyserver/caddy/v2"
+	"go.uber.org/zap"
+)
+
+// Interface guards
+var (
+	_ caddy.App         = (*App)(nil)
+	_ caddy.Module      = (*App)(nil)
+	_ caddy.Provisioner = (*App)(nil)
+)
+
+func init() {
+	caddy.RegisterModule(App{})
+}
+
+type App struct {
+	Supervise []Definition `json:"supervise,omitempty"`
+	log       *zap.Logger
+	supervisors []*Supervisor
+}
+
+// CaddyModule implements caddy.Module
+func (a App) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID:  "supervisor",
+		New: func() caddy.Module { return new(App) },
+	}
+}
+
+// Provision implements caddy.Provisioner
+func (a *App) Provision(context caddy.Context) error {
+	a.log = context.Logger(a)
+
+	for _, definition := range a.Supervise {
+
+		supervisors, err := definition.ToSupervisors(a.log)
+
+		if err != nil {
+			return err
+		}
+
+		a.log.
+			With(zap.Any("supervisors", supervisors)).
+			With(zap.Any("definition", definition)).
+			Debug("Supervisors created")
```

Comment:
This is a weird way to emit a log; I usually only use `With` if I'm re-assigning it to a variable so that multiple future log entries have those fields. If I were you I'd change them to `a.log.Debug("...", zap.Any(), zap.Any())`.